contracts/governance/src/collection.rs

```diff
@@ -0,0 +1,42 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use alloc::vec::Vec;
+use canonical::CanonError;
+use canonical_derive::Canon;
+
+#[derive(Clone, Canon, Debug)]
+pub struct Collection<K, V> {
+    data: Vec<(K, V)>,
+}
+
+impl<K: PartialEq, V> Collection<K, V> {
+    pub fn get(&self, key: &K) -> Result<Option<&V>, CanonError> {
+        Ok(self.data.iter().find(|(x, _)| x == key).map(|(_, y)| y))
+    }
+
+    pub fn insert(&mut self, key: K, value: V) -> Result<(), CanonError> {
+        self.data.push((key, value));
+
+        Ok(())
+    }
+
+    pub fn remove(&mut self, key: &K) -> Result<Option<V>, CanonError> {
+        if let Some(index) = self.data.iter().position(|(x, _)| x == key) {
+            let (_, val) = self.data.remove(index);
+
+            Ok(Some(val))
+        } else {
+            Ok(None)
+        }
```

Comment:
Consider using more performant options such as [Vec::retain](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.retain)

If we opt to go that way, we will have to update the signature and not return the removed element - which might not be a problem for our use case here

Reply:
the `remove` method has been consistent throughout the ecosystem with that signature. I'll change it since it doesn't fit out use case